src/files/routes.rs

```diff
@@ -49,7 +63,66 @@ async fn upload_file(
         }
     }
 
-    Ok((StatusCode::CREATED, Json(FileUploaded { id })))
+    Ok((StatusCode::CREATED, Json(FileWrapper { id })))
+}
+
+/// Axum service to handle file download requests
+/// FIXME: On integration, we need to do auth via token `Session` extrating on the request
+use axum_macros::debug_handler;
+#[debug_handler]
+async fn download_file(
+    // State(app): State<Arc<AppState>>,
+    // FIXME: harden extractor to only accept "*.filetype" or similar to prevent malicious client requests for "../../something.sh" and the like
+    Path(filename): Path<String>,
+    headers: HeaderMap,
+) -> Result<impl IntoApiResponse, AppError> {
+    let range = headers.get("range").ok_or(
+        AppError::new("Malformed file download request").with_status(StatusCode::NOT_ACCEPTABLE),
+    )?;
+
+    let filepath = format!("{}/{}", "/tmp", filename); // FIXME: use app state to set a file store location on disc
```

Comment:
The FIXME is misleading. the app state should probably have an implementation of a trait, where the current implementation would handing writing the file to local disk (but a future impl could be storing to a network store like S3).